src/main/java/uk/gov/ons/census/casesvc/model/dto/ModifiedAddress.java

```diff
@@ -6,11 +6,12 @@
 import lombok.Data;
 
 @Data
-@JsonInclude(Include.NON_ABSENT) // Only needed for marshalling (i.e. sending)
+@JsonInclude(Include.NON_NULL) // Only needed for marshalling (i.e. sending)
```

Comment:
I think `NON_ABSENT` was the right value, meaning "if it's not present, don't put it in the JSON" - that'll be needed for what we log, so that we know what was sent to us in the event

Reply:
you can put `NON_ABSENT` onto all the optionals, and `NON_NULL` onto the addressType maybe?

Reply:
I only saw the correct behaviour in the integration tests with `NON_NULL`. 

`NON_ABSENT` seems to not include empty optionals.

Reply:
I'm sure that `NON_ABSENT` is what we want - if you want your IT not to send a value, you might have to try with it null'ed. Let's chat on Slack about this

Reply:
Fundamentally, we shouldn't change the production code for the sake of the tests. If we need the tests to do something weird/different, we'll have to find a different way to build the messages. The most important thing is that the JSON payload persisted to the event log, matches the JSON that we received.

Reply:
As I understand it, that annotation is only there for the tests at the moment anyway? Our production code never marshals into this DTO. The problem I had with `NON_ABSENT` is it ignores both `null` values and `Optional.empty` values, giving me no way to distinguish between the json values being `null` and being not present in the tests. Happy to have a short conversation, it might be me doing the marshalling wrong.

Reply:
I was wrong! That'll learn me for not testing this for myself!